python-spec/src/somacore/data.py

```diff
@@ -58,14 +59,24 @@ def create(
         is provided, one will be added.  It may be used as an indexed column.
 
         :param uri: The URI where the ``DataFrame`` will be created.
+
         :param schema: Arrow schema defining the per-column schema. This schema
             must define all columns, including columns to be named as index
             columns.  If the schema includes types unsupported by the SOMA
             implementation, an error will be raised.
+
         :param index_column_names: A list of column names to use as user-defined
             index columns (e.g., ``['cell_type', 'tissue_type']``).
             All named columns must exist in the schema, and at least one
             index column name is required.
+
+        :param domains: An optional sequence of min-max tuples specifying the
```

Comment:
same comments as on the `tiledbsoma` PR review - the param description is confusing.  Suggest making it consistent with the other PR